src/Biryuchkov/Task5/main.cpp

```diff
@@ -0,0 +1,157 @@
+#include <QImage>
+#include <QColor>
+#include <QOpenGLFunctions>
+#include <QVector3D>
+#include <cmath>
+#include "Hit_ray.h"
+#include "Sphere.h"
+#include "Hit_light.h"
+#include "Surface.h"
+#include <memory>
+
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+constexpr auto WIDTH = 1280;
+constexpr auto HEIGHT = 960;
+constexpr auto FOV = M_PI/3.;
+constexpr auto MAX_REFLECTIONS = 4;
+constexpr auto BACKGROUND_COLOR = QVector3D(1.f, 1.0f, 0.7f);
+
+
+QVector3D mix(const QVector3D &a, const QVector3D &b, const float k){

```

Comment:
```suggestion
[[nodiscard]] auto mix(const QVector3D &a, const QVector3D &b, const float k) noexcept {

```

и в остальных функциях